cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamFileJanitor.java

```diff
@@ -115,4 +120,12 @@ void clean(StreamConfig config, long currentTime) throws IOException {
   private boolean isPartitionDirector(Location location) throws IOException {
     return (location.isDirectory() && location.getName().indexOf('.') > 0);
   }
+
+  private boolean isStreamExists(Id.Stream streamId) throws IOException {
```

Comment:
Right. 

I was only wondering why ``StreamAdmin.exists`` should throw anything other than an ``IOException``. I just checked that ``HBaseQueueAdmin.exists`` throws a ``DatasetManagementException``, so that's probably why.